script/tool/test/pubspec_check_command_test.dart

```diff
@@ -1743,7 +1743,11 @@ ${_topicsSection()}
         package.pubspecFile.writeAsStringSync('''
 ${_headerSection('a_package')}
 ${_environmentSection()}
-${_dependenciesSection(<String>['integration_test: \n    sdk: flutter'])}
+${_dependenciesSection(<String>[
+              'integration_test: \n    sdk: flutter',
+              'flutter_test: \n    sdk: flutter',
```

Comment:
If the existing tests are going to change to test more things, then the test descriptions need to change accordingly.

Reply:
updated test name to 'fails when integration_test, flutter_test or test are used in non dev dependency'